changelog.d/15074.bugfix

```diff
@@ -0,0 +1 @@
+Fix a long-standing bug where federated joins would fail if the first server in the list of servers to try is not in the room.
```

Comment:
Would this happen if you follow an outdated link with (now) incorrect `via` declared? I'm struggling to figure out the user impact.

Reply:
One might also have a valid `via`, but the first server in the list is temporarily offline.

Reply:
> One might also have a valid `via`, but the first server in the list is temporarily offline.

Oh, but that would result in a timeout and be handled differently by the joining server. I.e. I don't think this comment is relevant.

Reply:
> Would this happen if you follow an outdated link with (now) incorrect `via` declared? I'm struggling to figure out the user impact.

That's one way you can end up with a stale `via` list. Another way might be through space listings? I seem to remember a list of `via`s being stored in the space's state.

Reply:
Yes the children state events store vias in them that could get out of date.